readthedocs/proxito/views/serve.py

```diff
@@ -265,6 +265,14 @@ def _get_canonical_redirect_type(self, request):
                 log.debug("Proxito CNAME HTTPS Redirect.", domain=domain.domain)
                 return RedirectType.http_to_https
 
+        if (
+            unresolved_domain.is_from_public_domain
+            or unresolved_domain.is_from_external_domain
+        ):
+            if settings.PUBLIC_DOMAIN_USES_HTTPS and not request.is_secure():
+                # Redirect HTTP -> HTTPS (302) for public domains.
```

Comment:
Probably useful to have this comment above the `if` block, so it's more clear what each "block" is doing.